### DISCUSSION TITLE
Issues with rolling out Winget via Intune

### DISCUSSION THREAD

**Discussion by Folzus** [Weatherlights/Winget-AutoUpdate-Intune#26] [Q&A]
Hello dear community,

I want to roll out Winget via Intune on about 150 PCs. Before doing that I tested the rollout on several chosen devices. Some problems occured though that I haven't been able to fix so far. I hope you can maybe help me..

**Problem 1: Winget does not seem to update software properly on some devices**

After Winget was installed via intune, Winget searches for updates on logon (as expected). The problem is that winget tries to update the same software several times a day. The logs show that the update was successfull, however a couple of hours later it tries to do the same update again because the update was indeed not successful despite the contrary logs). I put the logs here:

[updates_1.log](https://github.com/Weatherlights/Winget-AutoUpdate-Intune/files/14123745/updates_1.log)

**Problem 2: Winget does not even try to update software on some devices**

Another problem is that on one device Winget is installed, but the update routine never triggers. I tried to trigger the routine in the task scheduler - but this didn't do anything (usually there would be notifications about each update in the notification center.) This is weird because the installation via intune happened under the same configurations across all machines. I put the logs here:

[updates_2.txt](https://github.com/Weatherlights/Winget-AutoUpdate-Intune/files/14123749/updates_2.txt)

Could the issue with problem 2 be that the device is not compliant with some policies? According to intune the computer risk level of that device is too high (I haven't figured out why so far but I am on it). 

I hope you could help me or give me a hint what could be wrong.

Thank you!

**Comment by Karl-WE** [comment]
@Folzus 
Could you provide the output of winget --version  from affected Clients, please?

**Comment by ne-tlw** [comment]
Problem 1: https://github.com/Weatherlights/Winget-AutoUpdate-Intune/issues/34
Problem 2: please check if winget is installed by executing winget as a logged-on user on the command line. If winget is missing on the client, Winget-AutoUpdate will not run (but it will log an error message to the logfile).
